packages/authentication/src/providers/google.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  AbstractAuthenticationModuleProvider,
+  AuthenticationResponse,
+} from "@medusajs/types"
+import { AuthProviderService, AuthUserService } from "@services"
+
+import { OAuth2 } from "oauth"
+import url, { UrlWithParsedQuery } from "url"
+import jwt, { JwtPayload } from "jsonwebtoken"
+import { MedusaError } from "@medusajs/utils"
+import { TreeLevelColumn } from "typeorm"
+
+type InjectedDependencies = {
+  authUserService: AuthUserService
+  authProviderService: AuthProviderService
+}
+
+type GoogleProviderConfig = {
+  callbackURL: string
+  clientID: string
+  clientSecret: string
+}
+
+class GoogleProvider extends AbstractAuthenticationModuleProvider {
```

Comment:
comment: As we discussed, the objective later on will be to have an AbstractOauthService that handle the heavy lifting. Will see at that moment the way we do the abstraction. But here it act as a first implementation so it is pushed for later